net-utils/src/lib.rs

```diff
@@ -252,6 +252,7 @@ pub struct SocketConfig {
 }
 
 impl SocketConfig {
+    #[deprecated(since = "2.3.0", note = "SO_REUSEPORT is now managed automatically")]
     pub fn reuseport(mut self, reuseport: bool) -> Self {
```

Comment:
This is the key change - we are making it such that users know that manually controlling reuseport flag is not supported anymore. This allows us to manipulate field in the SocketConfig as we deem appropriate.

Reply:
We would typically deprecate and then wait a version before removing.
I view overwriting the field as almost the same thing as just removing the API because the API becomes completely meaningless

Reply:
If someone keeps using this API, there are principally 2 outcomes: they are setting this field correctly for each of the functions they are calling, and our "rugging" is a noop, or they are setting it incorrectly, and "rugging" fixes a bug in their code. We can of course preserve the exact semantics here at the cost of complexity inside agave.

Reply:
Agree. "Fixing" their code is a behavioral breaking API change that probably makes life better in 99% of cases

Reply:
Now we are tracking whether user manually set the flag, and if they did we fall back to blindly obeying what they have specified.